Lib/test/test_grammar.py

```diff
@@ -251,6 +251,15 @@ def check_error(test):
         check("1e3in x")
         check("1jin x")
 
+        check("0xfnot in x")
+        check("0o7not in x")
+        check("0b1not in x")
+        check("9not in x")
+        check("0not in x")
+        check("1.not in x")
+        check("1e3not in x")
+        check("1jnot in x")
```

Comment:
Small nit: At this point maybe we want to make these strings dynamically:

```
for infix in ["and", "or", "in", "not in"]:
        check(f"0xf{infix} x")
        check(f"0o7{infix} x")
        check(f"0b1{infix} x")
        check(f"9{infix} x")
        check(f"0{infix} x")
        check(f"1.{infix} x")
        check(f"1e3{infix} x")
        check(f"1j{infix} x")
```

Reply:
I'll rewrite tests in a separate PR.